src/types.rs

```diff
@@ -1430,27 +1477,123 @@ impl Rule {
+    /// 
+    /// let mut sig = Signature::default();
+    /// let r = parse_rule(&mut sig, "A(x_) = B(x_) | C").expect("parse of A(x_) = B(x_) | C");
+    /// 
+    /// assert_eq!(r.len(), 2);
+    /// ```
     pub fn len(&self) -> usize {
         self.rhs.len()
     }
-    /// Is the rule empty?
+    /// Is the `Rule` empty?
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use term_rewriting::{Signature, Rule, Term, parse_term, parse_rule};
+    /// 
+    /// let mut sig = Signature::default();
+    /// let r = parse_rule(&mut sig, "A(x_) = B(x_) | C").expect("parse of A(x_) = B(x_) | C");
+    /// 
+    /// assert!(!r.is_empty());
+    /// 
+    /// let lhs = parse_term(&mut sig, "A").expect("parse of A");
+    /// let rhs : Vec<Term> = vec![];
+    /// let r = Rule::new(lhs, rhs).unwrap();
+    /// assert!(r.is_empty());
+    /// ```
     pub fn is_empty(&self) -> bool {
         self.rhs.is_empty()
     }
-    /// Give the lone RHS, if it exists
+    /// The lone RHS, if it exists. Otherwise, return `None`.
+    /// 
+    /// # Examples
+    ///
+    /// ```
+    /// # use term_rewriting::{Signature, Rule, Term, parse_term, parse_rule};
+    /// 
+    /// let mut sig = Signature::default();
+    /// let b = sig.new_op(0, Some("B".to_string()));
+    /// let r = parse_rule(&mut sig, "A = B").expect("parse of A = B");
+    ///
+    /// assert_eq!(
+    ///     r.rhs(), 
+    ///     Some(Term::Application { 
+    ///         op: b, 
+    ///         args: vec![], 
+    ///     }),
+    /// );
+    ///
+    /// let r = parse_rule(&mut sig, "A = B | C").expect("parse of A = B | C");
+    /// 
+    /// assert_eq!(r.rhs(), Option::None);
+    ///
+    /// let lhs = parse_term(&mut sig, "A").expect("parse of A");
+    /// let rhs = vec![];
+    /// let r = Rule::new(lhs, rhs).unwrap();
+    ///
+    /// assert_eq!(r.rhs(), Option::None);
+    /// ```
     pub fn rhs(&self) -> Option<Term> {
         if self.rhs.len() == 1 {
             Some(self.rhs[0].clone())
         } else {
             None
         }
     }
-    /// Return a list of the clauses in the `Rule`.
+    /// A list of the clauses in the `Rule`.
+    /// 
+    /// # Examples
+    ///
+    /// ```
+    /// # use term_rewriting::{Signature, Rule, Term, parse_term, parse_rule};
+    /// 
+    /// let mut sig = Signature::default();
+    ///
+    /// let r = parse_rule(&mut sig, "A = B").expect("parse of A = B");
+    ///
+    /// assert_eq!(r.clauses(), vec![r]);
+    ///
+    /// let r = parse_rule(&mut sig, "A = B | C").expect("parse of A = B | C");
+    /// let r1 = parse_rule(&mut sig, "A = B").expect("parse of A = B");
+    /// let r2 = parse_rule(&mut sig, "A = C").expect("parse of A = C");
+    /// 
+    /// assert_eq!(r.clauses(), vec![r1, r2]);
+    ///
+    /// let lhs = parse_term(&mut sig, "A").expect("parse of A");
+    /// let rhs = vec![];
+    /// let r = Rule::new(lhs, rhs).unwrap();
+    ///
+    /// assert_eq!(r.clauses(), vec![]);
```

Comment:
As above, I don't think we need to include examples of Rules with 0 RHSs unless there's some unexpected behavior specifically for that case.